### PR TITLE
True aaron vision sniper mode

### PR DESCRIPTION
**Pull request by WAaron-500**
Too lazy to fix manually just created new branch from main and went from there lmfao

### CODE REVIEW

File: src/main/java/frc/robot/commands/visioncommands/VisionSniperMode.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        xPos = Math.abs(xPos) > 1 ? xPos : 0;
+        SmartDashboard.putBoolean("Hub Visible", true);
+        SmartDashboard.putNumber("Hub Distance", Units.metersToFeet(FieldUtils.getHubDistance(yPos, hubCam)));
+
+        double turnSpeed = xPos * 0.0075;
+        double forwardSpeed = driveSpeed.get() * DRIVE_MULT;
+
+        drive.driveBase.curvatureDrive(-forwardSpeed, turnSpeed, true);
+    }
+
```
Comment (hadley31) [inline src/main/java/frc/robot/commands/visioncommands/VisionSniperMode.java:56]:
```suggestion

    @Override
    public void end() {
        hubCam.setLEDEnabled(false);
    }

```

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -194,6 +195,7 @@ private void configureButtonBindings() {
         TeleFeed feedCargoCommand = new TeleFeed(transversal, uptake, () -> 8.0);
         TeleFlyVar revFlywheelCommand = new TeleFlyVar(varFlyWheel);
 
+        VisionSniperMode RotateToHubAdjustable = new VisionSniperMode(hubCamera, drive, () -> controls.getLeftDriveY());
```
Comment (hadley31) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
        VisionSniperMode rotateToHubAdjustable = new VisionSniperMode(hubCamera, drive, () -> controls.getLeftDriveY());
```

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -224,7 +226,8 @@ private void configureButtonBindings() {
         controls.getIntakeRunInButton().whileActiveOnce(intakeInCommand);
         controls.getIntakeRunOutButton().whileActiveOnce(intakeOutCommand);
 
-        controls.getAutoAimButton().whileActiveOnce(rotateToHub);
+        controls.getAutoAimButton().whileActiveOnce(RotateToHubAdjustable);
```
Comment (hadley31) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
        controls.getAutoAimButton().whileActiveOnce(rotateToHubAdjustable);
```

**Review by hadley31 — CHANGES_REQUESTED** [review]
Looks good, minor changes requested

**Review by hadley31 — APPROVED** [review]
lgtm

**Comment by WAaron-500** [comment]
this is basically the other PR with less stuff so imma close it

**Pull request closed without merging**